packages/expo-gl/src/GLView.tsx

```diff
@@ -1,38 +1,44 @@
-// @flow
-
 import * as React from 'react';
 import PropTypes from 'prop-types';
 import { NativeModulesProxy, requireNativeViewManager } from 'expo-core';
 import { Platform, View, ViewPropTypes, findNodeHandle } from 'react-native';
 
-import packageJSON from '../package.json';
+const packageJSON = require('../package.json');
 
-import type { SurfaceCreateEvent, SnapshotOptions } from './GLView.types';
+import { SurfaceCreateEvent, ExpoWebGLRenderingContext, SnapshotOptions, GLViewProps } from './GLView.types';
+export { SurfaceCreateEvent, ExpoWebGLRenderingContext, SnapshotOptions, GLViewProps }
 
-type Props = {
-  /**
-   * Called when the OpenGL context is created, with the context object as a parameter. The context
-   * object has an API mirroring WebGL's WebGLRenderingContext.
-   */
-  onContextCreate?: (gl: *) => void,
+declare let global: any;
 
-  /**
-   * [iOS only] Number of samples for Apple's built-in multisampling.
-   */
-  msaaSamples: number,
+const { ExponentGLObjectManager, ExponentGLViewManager } = NativeModulesProxy;
+
+export interface NativeGLViewProps extends GLViewProps {
 
   /**
-   * A ref callback for the native GLView
-   */
-  nativeRef_EXPERIMENTAL: (React.ElementRef<typeof GLView.NativeView> | null) => void,
-} & React.ElementProps<typeof View>;
+  * Called when the OpenGL context is created, with the context object as a parameter. The context
+  * object has an API mirroring WebGL's WebGLRenderingContext.
+  */
+ onContextCreate(gl: ExpoWebGLRenderingContext): void;
+
+ /**
+  * [iOS only] Number of samples for Apple's built-in multisampling.
+  */
+ msaaSamples: number;
+
+ /**
+  * A ref callback for the native GLView
+  */
+ nativeRef_EXPERIMENTAL?(callback: ComponentOrHandle | null);
+}
 
-const { ExponentGLObjectManager, ExponentGLViewManager } = NativeModulesProxy;
+type ComponentOrHandle = null | number | React.Component<any, any> | React.ComponentClass<any>;
+const NativeView = requireNativeViewManager('ExponentGLView');
 
 /**
  * A component that acts as an OpenGL render target
  */
-export default class GLView extends React.Component<Props> {
+export default class GLView extends React.Component<NativeGLViewProps, { msaaSamples: number }> {
+  static NativeView: any;
```

Comment:
Do we need to expose this? It doesn't seem very useful, and is possibly misleading since "native views" from RN are strings at runtime.